src/executionFiles/switchChain.ts

```diff
@@ -0,0 +1,62 @@
+import { getChainById, SwitchChainHook } from '../types'
+import { Signer } from 'ethers'
+import StatusManager from '../StatusManager'
+import { Step } from '@lifinance/types'
+
+/**
+ * This method checks whether the signer is configured for the correct chain.
+ * If yes it returns the signer.
+ * If no and if user interaction is allowed it triggers the switchChainHook. If no user interaction is allowed it aborts.
+ *
+ * @param signer
+ * @param statusManager
+ * @param step
+ * @param switchChainHook
+ * @param allowUserInteraction
+ */
+export const switchChain = async (
+  signer: Signer,
+  statusManager: StatusManager,
+  step: Step,
+  switchChainHook: SwitchChainHook,
+  allowUserInteraction: boolean
```

Comment:
maybe default this to `false`?

Reply:
Why? it's not an optional value and right now we don't have a place where we don't want to pass a value to it. So it will always be defined